TSOClient/tso.simantics/NetPlay/EODs/Handlers/VMEODSlotsPlugin.cs

```diff
@@ -0,0 +1,678 @@
+                                machineType = 0;
+                                break;
+                            }
+                    }
+                }
+                State = VMEODSlotsStates.Managing;
+                int tempMachineBalance = (16384 * args[3]) + args[2];
+                userClient.Send("slots_owner_init", new byte[5] { Convert.ToByte(args[1]), Convert.ToByte(tempMachineBalance % 255),
+                    Convert.ToByte(tempMachineBalance / 255), Convert.ToByte(args[4]), machineType });
+            }
+        }
+        public override void OnDisconnection(VMEODClient client)
+        {
+            base.OnDisconnection(client);
+        }
+        private void ToggleOnOffHandler(string evt, string OnOffState, VMEODClient client)
+        {
+            onOffState = Int16.Parse(OnOffState);
+            client.SendOBJEvent(new VMEODEvent((short)VMOEDSlotsObjectEvents.ToggleOnOff));
+        }
+        private void NewOddsHandler(string evt, Byte [] args, VMEODClient client)
+        {
+            machineOdds = args[0];
+            client.SendOBJEvent(new VMEODEvent((short)VMOEDSlotsObjectEvents.SetNewPayout, new short[1] { machineOdds }));
+        }
+        private void WithdrawHandler(string evt, string amountString, VMEODClient client)
+        {
+            // parse the withdraw amount
+            short withdrawAmount = Int16.Parse(amountString);
+
+            // atempt to credit the owner by debiting the machine
+            var VM = client.vm;
+            VM.GlobalLink.PerformTransaction(VM, false, Server.Object.PersistID, client.Avatar.PersistID, Convert.ToInt32(withdrawAmount),
+
+            (bool success, int transferAmount, uint uid1, uint budget1, uint uid2, uint budget2) =>
+            {
+                    //TODO: Make this part of global link
+                    VM.SendCommand(new VMNetAsyncResponseCmd(0, new VMTransferFundsState
+                {
+                    Responded = true,
+                    Success = success,
+                    TransferAmount = transferAmount,
+                    UID1 = uid1,
+                    Budget1 = budget1,
+                    UID2 = uid2,
+                    Budget2 = budget2
+                }));
+                if (success)
+                {
+                    client.Send("slots_resume_manage", amountString);
+                }
+                else
+                {
+                    Console.WriteLine("VMEODSlotsPlugin.WithdrawHandler: There was an error trying to withdraw money to the owner!");
+                    client.Send("slots_resume_manage", "0");
+                }
+            });
+
+            // client.SendOBJEvent(new VMEODEvent((short)VMOEDSlotsObjectEvents.WithdrawCash)); *Depricated*
+        }
+        private void DepositHandler(string evt, string amountString, VMEODClient client)
+        {
+            // parse the withdraw amount
+            short depositAmount = Int16.Parse(amountString);
```

Comment:
Ditto but in reverse. A negative value could withdraw money from the machine. (plus needs machine owner check)